I get a build failure when building on master: https://paste.fliegendewurst.eu/LyBtYd.log
```
FAILED: libastroterm_lib.a.p/src_core.c.o 
gcc -Ilibastroterm_lib.a.p -I. -I.. -I../include -Idata -I../data -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -DHAVE_ARGTABLE3 -fPIC -MD -MQ libastroterm_lib.a.p/src_core.c.o -MF libastroterm_lib.a.p/src_core.c.o.d -o libastroterm_lib.a.p/src_core.c.o -c ../src/core.c
../src/core.c: In function 'string_to_time':
../src/core.c:493:27: error: implicit declaration of function 'strptime'; did you mean 'strftime'? [-Wimplicit-function-declaration]
  493 |     const char *pointer = strptime(string, "%Y-%m-%dT%H:%M:%S", time);
      |                           ^~~~~~~~
      |                           strftime
../src/core.c:493:27: error: initialization of 'const char *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
```

--- a/pkgs/by-name/as/astroterm/package.nix
+++ b/pkgs/by-name/as/astroterm/package.nix
@@ -1,0 +1,56 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  fetchFromGitHub,
+  xxd,
+  meson,
+  ninja,
+  ncurses,
+  argtable,
+  versionCheckHook,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "astroterm";
+  version = "1.0.4";
+
+  src = fetchFromGitHub {
+    owner = "da-luce";
+    repo = "astroterm";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-CYKW/RAQ3a5238cojbpGfTenMQApfaZOHnQMrZ6LWzA=";
+  };
+
+  bsc5File = fetchurl {
+    url = "https://web.archive.org/web/20231007085824/http://tdc-www.harvard.edu/catalogs/BSC5";
+    hash = "sha256-5HHQLq9O7LYcEvh5octkMrqde2ipqMVlSh60KgyMw0A=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    xxd
+    versionCheckHook
+  ];
+  buildInputs = [
+    argtable
+    ncurses
+  ];
+
+  postPatch = ''
+    mkdir -p data
+    ln -s ${finalAttrs.bsc5File} data/bsc5
+  '';
+
+  doCheck = true;
+
+  meta = {
+    description = "Celestial viewer for the terminal, written in C";
+    homepage = "https://github.com/da-luce/astroterm/";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.da-luce ];
+    mainProgram = "astroterm";
+    platforms = lib.platforms.unix;
+  };
+})

